include/util/indexed_data.hpp

```diff
@@ -107,6 +108,11 @@ template <int N, typename T = std::string> struct VariableGroupBlock
 
         out.write((const char *)&refernce, sizeof(refernce));
 
+        if (!out.good())
+        {
+            throw util::exception(std::string("Error writing block reference: ") + SOURCE_REF);
+        }
```

Comment:
re. `stream.good()` vs `stream.fail()` - if you check the table at the bottom 

http://en.cppreference.com/w/cpp/io/basic_ios/good

you will see that `fail` _is not_ the negation of `good`. It behaves differently for the eof bit. `.good` returns false if the eofbit is set, `.fail` returns false there too (!). I would keep this in mind and simply always check streams via `if (!stream)`.